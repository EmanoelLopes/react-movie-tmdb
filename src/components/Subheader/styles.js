import styled from 'styled-components';
import { colors } from 'utils/styled';

export const SubheaderWrapper = styled.section`
  background-color: ${colors.movieDBColor};
  color: ${colors.mainColor};
  display: flex;
  justify-content: center;
  padding: 15px 0;

  .container {
    justify-content: space-between;
  }

  .navigation {
    > a {
      color: ${colors.mainColor};
      font-size: 18px;
      margin-right: 15px;
      text-decoration: none;
    }
  }

  select {
    background-color: ${colors.movieDBColor};
    border: none;
    font-size: 15px;
    max-width: 130px;
  }
`;
