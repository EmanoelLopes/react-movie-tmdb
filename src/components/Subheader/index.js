import React from 'react';
import { Link } from 'react-router-dom';
import SelectLanguage from 'components/SelectLanguage';
import { SubheaderWrapper } from './styles';  

const Subheader = () => {
  return (
    <SubheaderWrapper>
      <div className="container">
        <div className="navigation">
          <Link to='/movies'>Movies</Link>
          <Link to='/tvs'>TV Series</Link>
        </div>
        <SelectLanguage />
      </div>
    </SubheaderWrapper>
  );
};

export default Subheader;
