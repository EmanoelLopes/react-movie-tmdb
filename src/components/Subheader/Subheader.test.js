import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter as RouterWrapper } from 'react-router-dom';
import Subheader from 'components/Subheader';
import locale from 'lang';

locale.init();

let wrapped;
let defaultProps = {};

beforeEach(() => {
  wrapped = mount(
    <RouterWrapper>
      <Subheader {...defaultProps} />
    </RouterWrapper>
  );
});

afterEach(() => {
  wrapped.unmount();
});

describe('<Subheader />', () => {
  it('should match snapshot', () => {
    expect(wrapped).toMatchSnapshot();
  });
});
