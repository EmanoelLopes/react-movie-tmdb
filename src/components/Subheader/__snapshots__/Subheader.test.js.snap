// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Subheader /> should match snapshot 1`] = `
.c0 {
  background-color: #01d277;
  color: #081c24;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 15px 0;
}

.c0 .container {
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 .navigation > a {
  margin-right: 15px;
  color: #081c24;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Subheader>
      <styled.section>
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bwzfXH",
                "isStatic": true,
                "lastClassName": "c0",
                "rules": Array [
                  "
  background-color: ",
                  "#01d277",
                  ";
  color: ",
                  "#081c24",
                  ";
  display: flex;
  justify-content: center;
  padding: 15px 0;

  .container {
    justify-content: space-between;
  }

  .navigation {
    > a {
      margin-right: 15px;
      color: ",
                  "#081c24",
                  ";
      text-decoration: none;
    }
  }
",
                ],
              },
              "displayName": "styled.section",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bwzfXH",
              "target": "section",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <section
            className="c0"
          >
            <div
              className="container"
            >
              <div
                className="navigation"
              >
                <Link
                  replace={false}
                  to="/movies"
                >
                  <a
                    href="/movies"
                    onClick={[Function]}
                  >
                    Movies
                  </a>
                </Link>
                <Link
                  replace={false}
                  to="/tvs"
                >
                  <a
                    href="/tvs"
                    onClick={[Function]}
                  >
                    TV Series
                  </a>
                </Link>
              </div>
              <SelectLanguage>
                <select
                  defaultValue={null}
                  onChange={[Function]}
                >
                  <option
                    disabled={true}
                    key="0"
                    value="none"
                  >
                    Choose language
                  </option>
                  <option
                    disabled={false}
                    key="1"
                    value="pt-BR"
                  >
                    Português (BR)
                  </option>
                  <option
                    disabled={false}
                    key="2"
                    value="es-ES"
                  >
                    Español
                  </option>
                  <option
                    disabled={false}
                    key="3"
                    value="en-US"
                  >
                    English
                  </option>
                </select>
              </SelectLanguage>
            </div>
          </section>
        </StyledComponent>
      </styled.section>
    </Subheader>
  </Router>
</BrowserRouter>
`;
